src/algebra/order/floor.lean

```diff
@@ -604,6 +607,18 @@ lemma floor_to_nat (a : α) : ⌊a⌋.to_nat = ⌊a⌋₊ := rfl
 
 lemma ceil_to_nat  (a : α) : ⌈a⌉.to_nat = ⌈a⌉₊ := rfl
 
+lemma coe_nat_floor_eq_floor {a : α} (ha : 0 ≤ a) : (⌊a⌋₊ : ℤ) = ⌊a⌋ :=
```

Comment:
What do you think of naming this `nat.cast_floor_eq_int_floor` to keep the `nat.cast` atom? THen your current `int.coe_nat_floor_eq_floor'` can be renamed to `nat.cast_floor_eq_cast_int_floor`.